src/Core/CoreKernel.php

```diff
@@ -130,12 +150,56 @@ protected function bootDatabaseEnvVars()
      * @return array
      */
     protected function getDatabaseConfig()
+    {
+        return $this->getSingleDataBaseConfig(0);
+    }
+
+    private function getDatabaseReplicaConfig(int $replica)
+    {
+        return $this->getSingleDataBaseConfig($replica);
+    }
```

Comment:
Why not just do this?
```suggestion
    protected function getDatabaseConfig(int $replica = 0)
    {
        return $this->getSingleDataBaseConfig($replica);
    }
```

Reply:
If not doing that, you should update `getDatabaseReplicaConfig` so that passing `0` throws an exception, since `0` represents the primary, not a replica

Reply:
Updated